packages/block-library/src/form/index.php

```diff
@@ -0,0 +1,108 @@
+<?php
+/**
+ * Server-side rendering of the `core/form` block.
+ *
+ * @package WordPress
+ */
+
+/**
+ * Renders the `core/form` block on server.
+ *
+ * @param array    $attributes The block attributes.
+ * @param string   $content The saved content.
+ * @param WP_Block $block The parsed block.
+ *
+ * @return string The content of the block being rendered.
+ */
+function render_block_core_form( $attributes, $content, $block ) {
+
+	// Build the form action attribute.
+	$action = empty( $attributes['action'] ) ? '' : $attributes['action'];
+	$action = empty( $action ) ? '#' : $action;
+
+	/**
+	 * Filters the form action attribute.
+	 *
+	 * @param string $action The form action attribute.
+	 * @param array  $attributes The block attributes.
+	 * @param string $content The saved content.
+	 * @param WP_Block $block The parsed block.
+	 *
+	 * @return string The form action attribute.
+	 */
+	$action = apply_filters( 'block_form_action', $action, $attributes, $content, $block );
+
+	// Build the form method attribute.
+	$method = empty( $attributes['method'] ) ? '' : $attributes['method'];
+	$method = empty( $method ) || ( 'get' !== strtolower( $method ) && 'post' !== strtolower( $method ) ) ? 'post' : $method;
+	/**
+	 * Filters the form method attribute.
+	 *
+	 * @param string $method The form method attribute.
+	 * @param array  $attributes The block attributes.
+	 * @param string $content The saved content.
+	 * @param WP_Block $block The parsed block.
+	 *
+	 * @return string The form method attribute.
+	 */
+	$method = apply_filters( 'block_form_method', $method, $attributes, $content, $block );
+
+	$form_id = empty( $attributes['formId'] ) ? md5( json_encode( array( $attributes, $block ) ) ) : $attributes['formId'];
+
+	return str_replace(
+		array(
+			'<form ',
```

Comment:
This is a great usecase for the new HTML helper using nexttag?

Reply:
Good point. Done 👍